tests/api/test_category.py

```diff
@@ -19,7 +19,7 @@ def test_category_query(user_api_client, product):
         category(id: "%(category_pk)s") {
             id
             name
-            ancestors(last: 20) {
+            ancestors(first: 20) {
```

Comment:
Why did you change last to first?

Reply:
I think it was part of @patrys  testing when he started this PR

Reply:
I can revert this change

Reply:
No, it's no longer legal to specify `last` before specifying `before`.

Reply:
actually I talked (i think with @maarcingebala if he can confirm) that you can specify `last` without `before` becuase we can actually get last records (it's pointless because it same results as first but with revert ordering)

Reply:
There is already an option to reverse the results, I don't think it makes sense to support `last` without `before`.

Reply:
but when you won't provide sort_by arg then you cannot reverse results
this doesn't add complicity in code or anything
we support only last as default
I don't see a point for removing support for `last` now

Reply:
That's interesting, why can't you reverse it then?

Reply:
when you send query like this
`
query {
customers(first 100){
 name
}
}
`

you are unable to reverse it's order
you have to provide sorter_by
so you can use ASC DESC

Reply:
Isn't `first: 100` the same as `first: 100, sortBy: {field: SOMETHING}`? And if not, _why not_?

Reply:
not exactly

only if `SOMETHING` is default sorting

in that case
`first: 100` the same as `first: 100, sortBy: {field: SOMETHING, direction: ASC}`
and
`last: 100` the same as `first: 100, sortBy: {field: SOMETHING, direction: DESC}`

Reply:
>`last: 100` the same as `first: 100, sortBy: {field: SOMETHING, direction: DESC}`

I think they are actually not the same. I would expect `last` without a `before` to return the correct items (eg. the same as `first` with `direction: DESC`) but in the wrong order (the entire list reversed).

Consider nine products with names A, B, C, D, E, F, G, H, I.

* `first: 3`, `direction: ASC` -> `["A", "B", "C"]`
* `last: 3`, `direction: ASC` -> `["G", "H", "I"]`
* `first: 3`, `direction: DESC` -> `["I", "H", "G"]`
* `last: 3`, `direction: DESC` -> `["G", "H", "I"]`

So it's not a convenient way to reverse the sorting order. I don't see a reason to support this.

Reply:
ok, you're right, I will add constraint on `before` :P